samples/crypto/ecjpake/src/main.c

```diff
@@ -0,0 +1,261 @@
+			LOG_INF("psa_pake_output failed. (Error: %d)", status);
+			return status;
+		}
+
+		status = psa_pake_input(client, round1_steps[i], buffer, outlen);
+		if (status != PSA_SUCCESS) {
+			LOG_INF("psa_pake_input failed. (Error: %d)", status);
+			return status;
+		}
+	}
+
+	/* Client provides data to server. */
+	for (uint32_t i = 0; i < ARRAY_SIZE(round1_steps); i++) {
+		status = psa_pake_output(client, round1_steps[i], buffer, sizeof(buffer), &outlen);
+		if (status != PSA_SUCCESS) {
+			LOG_INF("psa_pake_output failed. (Error: %d)", status);
+			return status;
+		}
+
+		status = psa_pake_input(server, round1_steps[i], buffer, outlen);
+		if (status != PSA_SUCCESS) {
+			LOG_INF("psa_pake_input failed. (Error: %d)", status);
+			return status;
+		}
+	}
+
+	psa_pake_step_t round2_steps[] = { PSA_PAKE_STEP_KEY_SHARE, PSA_PAKE_STEP_ZK_PUBLIC,
+					   PSA_PAKE_STEP_ZK_PROOF };
+
+	/* Server provides data to client */
+	for (uint32_t i = 0; i < ARRAY_SIZE(round2_steps); i++) {
+		status = psa_pake_output(server, round2_steps[i], buffer, sizeof(buffer), &outlen);
+		if (status != PSA_SUCCESS) {
+			LOG_INF("psa_pake_output failed. (Error: %d)", status);
+			return status;
+		}
+
+		status = psa_pake_input(client, round2_steps[i], buffer, outlen);
+		if (status != PSA_SUCCESS) {
+			LOG_INF("psa_pake_input failed. (Error: %d)", status);
+			return status;
+		}
+	}
+
+	/* Client provides data to server. */
+	for (uint32_t i = 0; i < ARRAY_SIZE(round2_steps); i++) {
+		status = psa_pake_output(client, round2_steps[i], buffer, sizeof(buffer), &outlen);
+		if (status != PSA_SUCCESS) {
+			LOG_INF("psa_pake_output failed. (Error: %d)", status);
+			return status;
+		}
+
+		status = psa_pake_input(server, round2_steps[i], buffer, outlen);
+		if (status != PSA_SUCCESS) {
+			LOG_INF("psa_pake_input failed. (Error: %d)", status);
+			return status;
+		}
+	}
+
+	return status;
+}
+
+int main(void)
+{
+	psa_status_t status = psa_crypto_init();
+
+	if (status != PSA_SUCCESS) {
+		LOG_INF("psa_crypto_init failed. (Error: %d)", status);
+	}
+
+	psa_pake_cipher_suite_t cipher_suite = PSA_PAKE_CIPHER_SUITE_INIT;
+
+	psa_pake_cs_set_algorithm(&cipher_suite, PSA_ALG_JPAKE);
+	psa_pake_cs_set_primitive(&cipher_suite, PSA_PAKE_PRIMITIVE(PSA_PAKE_PRIMITIVE_TYPE_ECC,
+								    PSA_ECC_FAMILY_SECP_R1, 256));
+	psa_pake_cs_set_hash(&cipher_suite, PSA_ALG_SHA_256);
+
+	psa_key_attributes_t key_attributes = PSA_KEY_ATTRIBUTES_INIT;
+
+	psa_set_key_usage_flags(&key_attributes, PSA_KEY_USAGE_DERIVE);
+	psa_set_key_algorithm(&key_attributes, PSA_ALG_JPAKE);
+	psa_set_key_type(&key_attributes, PSA_KEY_TYPE_PASSWORD);
+
+	psa_key_id_t key;
+
+	status = psa_import_key(&key_attributes, "password", 8, &key);
+	if (status != PSA_SUCCESS) {
+		LOG_INF("psa_import_key failed. (Error: %d)", status);
+	}
+
+	/* Initialize PAKE operation object for the client.*/
+	psa_pake_operation_t client = PSA_PAKE_OPERATION_INIT;
+
+	status = psa_pake_setup(&client, &cipher_suite);
```

Comment:
The main function is a bit difficult to read and it has some code duplication.
For example from this line  until the call to the psa_pake_set_password_key()

The process is identical and the only differently is the referenced objects and the ids. We can create a function to make it reduce the code and make it more clear. 

Overall I propose to split the functionality to functions  so that  main just calls to functions which perform different steps of the process.

Reply:
Not sure if it makes things clearer, given the size and purpose of the sample, but I can do it.